pkg/proxy/iptables/proxier.go

```diff
@@ -591,6 +593,58 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 	proxier.syncProxyRules()
 }
 
+// OnNodeAdd is called whenever creation of new node object
+// is observed.
+func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
+	if node.Name != proxier.hostname {
+		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		return
+	}
+	new := node.Labels
+	old := proxier.nodeLabels
+	proxier.mu.Lock()
+	proxier.nodeLabels = new
+	proxier.mu.Unlock()
+	if len(old) != len(new) || !reflect.DeepEqual(old, new) {
```

Comment:
length is already considered from reflecy.DeepEqual?

Reply:
Yes, gonna remove length comparation.